### PR TITLE
Feature: Flags to opt-out forced updates

### PR DESCRIPTION
**Pull request by carlosjgp**
## Use case:

Some of the official helm charts we deploy have secrets generated from their `values.yaml` and every time we run landscaper without changing anything on those deployments or environment variables

The charts that are currently behaving in this way are `stable/logstash` and `stable/rabbitmq`

## Implemented feature:

Opt-out forced updates using the flag `--disable-forced-updates` or only dsiable forced updates for secret changes `--disable-secrets-forced-updates`

### CODE REVIEW

File: pkg/landscaper/executor.go [hunk 1]
```diff
@@ -48,18 +52,20 @@ func NewExecutor(helmClient helm.Interface, chartLoader ChartLoader, kubeSecrets
 func (e *executor) gatherForcedUpdates(current, update Components) (map[string]bool, error) {
 	needForcedUpdate := map[string]bool{}
 
-	for _, cmp := range update {
-		// releases that differ only in secret values are forced so that pods will restart with the new values
-		for _, curCmp := range current {
-			if curCmp.Name == cmp.Name && isOnlySecretValueDiff(*curCmp, *cmp) {
-				logrus.Infof("%s differs in secrets values only; don't update but delete + create instead", cmp.Name)
-				needForcedUpdate[cmp.Name] = true
+	if e.disabledForcedUpdates == false {
```
Comment (rollulus) [inline pkg/landscaper/executor.go]: What about something like:
```
// return early
if e.disabledForcedUpdates { return needForcedUpdate, nil }
// normal flow
for _, cmp := ... {
```
I believe some Go celebrity once wrote something about this to give it a more weight than just my taste, but I cannot find it.

Comment (rollulus) [reply pkg/landscaper/executor.go]: And I just stumbled upon a credible source: 
https://dave.cheney.net/practical-go/presentations/qcon-china.html#_return_early_rather_than_nesting_deeply

Comment (carlosjgp) [reply pkg/landscaper/executor.go]: I'll take a look soon! I'm currently moving to a new place 😅

Comment (rollulus) [reply pkg/landscaper/executor.go]: No problem, good luck man!

**Comment by eimarfandino** [comment]
i think this is already resolved by https://github.com/Eneco/landscaper/pull/123

**Comment by rollulus** [comment]
Alright, thanks for letting me know @eimarfandino.

**Pull request closed without merging**